pkg/controller/containersource/reconcile.go

```diff
@@ -133,29 +135,46 @@ func (r *reconciler) getSinkUri(ctx context.Context, source *v1alpha1.ContainerS
 		return "", fmt.Errorf("sink ref is nil")
 	}
 
-	obj, err := r.fetchObjectReference(source.Namespace, source.Spec.Sink)
+	obj, err := r.fetchObjectReference(ctx, source.Namespace, source.Spec.Sink)
 	if err != nil {
-		logger.Warnf("Failed to fetch sink target %+v: %s", source.Spec.Sink, err)
+		logger.Warnf("Failed to fetch sink target %+v: %s", source.Spec.Sink, zap.Error(err))
 		return "", err
 	}
 	t := duckv1alpha1.Sink{}
 	err = duck.FromUnstructured(obj, &t)
 	if err != nil {
-		logger.Warnf("Failed to deserialize sink: %s", err)
+		logger.Warnf("Failed to deserialize sink: %s", zap.Error(err))
 		return "", err
 	}
 
 	if t.Status.Sinkable != nil {
-		return t.Status.Sinkable.DomainInternal, nil
+		return fmt.Sprintf("http://%s/", t.Status.Sinkable.DomainInternal), nil
```

Comment:
Strange that http is hardcoded here. Is cluster-internal traffic always https capable?

Reply:
We will have to revisit this when we decide how we will support paths. But at the moment, all domains == http endpoint on port :80 from what I can see. Let me know if this is incorrect.